parquet/src/main/java/org/apache/iceberg/data/parquet/BaseParquetWriter.java

```diff
@@ -190,33 +210,49 @@ public Optional<ParquetValueWriters.PrimitiveWriter<?>> visit(
     }
 
     @Override
-    public Optional<ParquetValueWriters.PrimitiveWriter<?>> visit(
+    public Optional<ParquetValueWriter<?>> visit(
         LogicalTypeAnnotation.DateLogicalTypeAnnotation dateType) {
-      return Optional.of(new DateWriter(desc));
+      ParquetValueWriter<?> dateWriter = dateWriter(desc);
+      if (dateWriter != null) {
```

Comment:
There is no need to handle nulls here. A writer should always be returned by the factory methods.